src/coreclr/jit/emitxarch.cpp

```diff
@@ -6267,6 +6272,11 @@ void emitter::emitIns_SIMD_R_R_R_R(
 
         emitIns_R_R_R(ins, attr, targetReg, op2Reg, op3Reg);
     }
+    else if (IsAVXVNNIInstruction(ins))
+    {
+        assert(UseVEXEncoding());
+        emitIns_R_R_R(ins, attr, targetReg, op2Reg, op3Reg);
```

Comment:
Shouldn't this go down the same path as `IsFMAInstruction` so the `movaps` is emitted if `op1Reg != targetReg`?

Reply:
In some cases, the targetReg would be equal to op3Reg. This is also the reason I set targetReg to op1Reg.

Reply:
That should be covered by setting op2Reg/op3Reg as delay free.

Lowering should be checking and marking any operands that can be memory operands as either `contained` or `regOptional` (assuming its not the target, since that requires handling via `STORE_*` nodes instead). That is, if `op3` can be `xmm` or `[mem]`, then you would have something like the following in https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/lowerxarch.cpp#L5596:
```cpp
if (IsContainableHWIntrinsicOp(node, op3, &supportsRegOptional))
{
    MakeSrcContained(node, op3);
}
else if (supportsRegOptional)
{
    op3->SetRegOptional();
}
```

Then, in the registering allocator you likely need something like (swapping the `op3->isContained()` check with the appropriate node that can be a memory operand or removing it if none can be a memory operand):
```cpp
tgtPrefUse = BuildUse(op1);
srcCount += 1;
srcCount += BuildDelayFreeUses(op2, op1);
srcCount += op3->isContained() ? BuildOperandUses(op3) : BuildDelayFreeUses(op3, op1);
```

This will cause the register allocator to prefer `tgt == op1` (this isn't guaranteed, which is what the `movaps` covers) and enforces that `op2 != tgt && op3 != tgt`

@kunalspathak should be able to give additional input/considerations as appropriate.